### DISCUSSION TITLE
For classification problems, should model always have logits output for loss calculation?

### DISCUSSION THREAD

**Discussion by Ahnaf19** [mrdbourke/pytorch-deep-learning#1096] [Q&A]
I am a bit confused: previous of starting this course, I saw models with softmax, sigmoid or `torch.round()` at the end of the layer for classification tasks, which is understandable to me. But, here (I am at the end of 03. PyTorch Computer Vision) I understood about logits. It seems like we calculate logits --> prob_preds --> pred_labels.

Should my model always have logits output for loss calculation and apply softmax/sigmoid separately for getting pred_labels?
